code/modules/mob/living/carbon/xenomorph/castes/hivemind/hivemind.dm

```diff
@@ -46,12 +46,16 @@
 	return ..()
 
 /mob/living/carbon/xenomorph/hivemind/Move(NewLoc, Dir = 0)
-	var/obj/effect/alien/weeds/W = locate() in range("3x3", NewLoc)
+	var/obj/effect/alien/weeds/W = locate() in range(1, NewLoc)
 	if(!W)
-		var/obj/effect/alien/weeds/nearby = locate() in range("3x3", loc)
+		var/obj/effect/alien/weeds/nearby = locate() in range(1, loc)
 		if(!nearby)
-			// If we run out of weeds just teleport to some random weeds.
-			forceMove(get_turf(core))
+			var/obj/effect/alien/weeds/nearbiest = locate() in  range(7, loc)
+			if(nearbiest) 	// If we run out of weeds just teleport to some random weeds that aint so far away
+				forceMove(get_turf(nearbiest))
+			else
+				forceMove(get_turf(core))
```

Comment:
Not a fan of this locate() spam Can you just revert it to previous functionality here

Reply:
sure